apps/balancer-tools/seed.ts

```diff
@@ -523,83 +592,98 @@ DO UPDATE SET chain_id = EXCLUDED.chain_id, updated_at = NOW();
-// FROM
-// 	pool_tokens pt
-// 	INNER JOIN pool_snapshots ps ON pt.pool_external_id = ps.pool_external_id
-// 	LEFT JOIN token_prices tp ON pt.token_address = tp.token_address
-// 		AND pt.network_slug = tp.network_slug
-// 		AND date_trunc('day', ps.timestamp) = date_trunc('day', tp.timestamp)
-// WHERE
-// 	tp.id IS NULL
-// ORDER BY
-// 	day,
-// 	pt.token_address;
-//   `);
-
-//   // Step 2: Deduplicate tokens for the same day
-//   const dedupedTokens: { [day: string]: Set<string> } = {};
-
-//   for (const row of result) {
-//     const day = row.day.toISOString();
-//     if (!dedupedTokens[day]) {
-//       dedupedTokens[day] = new Set();
-//     }
-
-//     dedupedTokens[day].add(
-//       `${remappings[row.network_slug] ?? row.network_slug}:${row.token_address}`
-//     );
-//   }
-
-//   // Step 3: Fetch token prices
-//   for (const [day, tokens] of Object.entries(dedupedTokens)) {
-//     const dateTimestamp = Date.UTC(
-//       Number(day.split("-")[0]),
-//       Number(day.split("-")[1]) - 1,
-//       Number(day.split("-")[2].split("T")[0])
-//     );
-//     const tokenAddresses = Array.from(tokens);
-//     try {
-//       const prices = await DefiLlamaAPI.getHistoricalPrice(
-//         new Date(dateTimestamp),
-//         tokenAddresses
-//       );
-
-//       const entries = Object.entries(prices.coins);
-
-//       await addToTable(
-//         tokenPrices,
-//         entries.map((entry) => ({
-//           tokenAddress: entry[0].split(":")[1],
-//           priceUSD: entry[1].price,
-//           timestamp: new Date(entry[1].timestamp * 1000),
-//           networkSlug:
-//             // @ts-ignore
-//             inverseRemapping[entry[0].split(":")[0]] ?? entry[0].split(":")[0],
-//         }))
-//       );
-
-//       console.log(`Fetched prices for tokens on ${day}:`, prices);
-//     } catch (e) {
-//       console.error(
-//         `Failed to fetch prices for tokens on ${day}: ${e.message}`
-//       );
-//     }
-//   }
-// }
+async function seedVebalRounds() {
+  logIfVerbose("Seeding veBAL rounds");
+
+  const startDate = new Date("2022-04-14T00:00:00.000Z");
+  let roundNumber = 1;
+
+  while (startDate <= new Date()) {
+    const endDate = new Date(startDate);
+    endDate.setUTCDate(endDate.getUTCDate() + 6);
+    endDate.setUTCHours(23, 59, 59, 999);
+
+    await db.execute(sql`
+      INSERT INTO vebal_rounds (start_date, end_date, round_number)
+      VALUES (${startDate.toISOString()}, ${endDate.toISOString()}, ${roundNumber})
+      ON CONFLICT (round_number) DO NOTHING;
+    `);
+    startDate.setUTCDate(startDate.getUTCDate() + 7);
+    roundNumber++;
+  }
+}
+
+async function seedBalEmission() {
+  const timestamps = await db
+    .selectDistinct({ timestamp: poolSnapshots.timestamp })
+    .from(poolSnapshots);
+
+  for (const { timestamp } of timestamps) {
+    if (!timestamp) continue;
+    try {
+      const weeklyBalEmission = balEmissions.weekly(dateToEpoch(timestamp));
```

Comment:
@mendesfabio used to logic we already have on JS to not take so much time to get the query right